### ISSUE TITLE
Please provide example in test having System.out redirected to a JediTermWidget

### ISSUE THREAD

**Issue by michael-newsrx** [JetBrains/jediterm#188]
I've been trying to figure out how to take an OutputStream and send it to a JediTermWidget for display.

I've been through the test and other code and have failed to figure this out.

Ideally I'd like an example that would be equivalent in functionality to
```Java
		SettingsProvider settingsProvider=new DefaultSettingsProvider();
		JediTermWidget tw = new JediTermWidget(settingsProvider);		
		JFrame jframe = new JFrame();
		jframe.setSize(1280,720);
		jframe.add(tw);
		jframe.setVisible(true);

		TeeStream tee_stdout = new TeeStream(System.out, tw.getOutputStream());
		TeeStream tee_stderr = new TeeStream(System.err, tw.getOutputErrorStream());

		System.setOut(tee_stdout);
		System.setErr(tee_stderr);

		System.out.println("HELLO"); //output should show up in terminal widget
```

**Comment by michael-newsrx** [comment]
FYI: I'm pulling in jediterm via jitpack:

```Gradle
    compile 'com.github.JetBrains:jediterm:ca8db6c4c4'
    compile 'log4j:log4j:1.2.17'
    compile 'org.jetbrains:annotations:13.0'
    compile 'com.google.guava:guava:25.1-jre'
```

**Comment by michael-newsrx** [comment]
I figured out a way (probably wrong) to get System.out redirected into a self hosted terminal, but trying to use ANSI escape sequences to set colors, clear screen, etc, aren't working? What am I doing wrong here?

```Java
package tests.manual;

import java.io.IOException;
import java.io.OutputStream;
import java.io.PrintStream;

import javax.swing.JFrame;

import com.jediterm.terminal.ArrayTerminalDataStream;
import com.jediterm.terminal.TerminalMode;
import com.jediterm.terminal.emulator.JediEmulator;
import com.jediterm.terminal.ui.JediTermWidget;
import com.jediterm.terminal.ui.settings.DefaultSettingsProvider;
import com.jediterm.terminal.ui.settings.SettingsProvider;
import com.newsrx.gui.TeeStream;

public class JediTerminal {

	private static final char ESC = 27;

	public static void main(String[] args) throws InterruptedException, IOException {
		termInit();
	}

	private static void termInit() throws InterruptedException, IOException {

		SettingsProvider settingsProvider = new DefaultSettingsProvider() {
			@Override
			public float getTerminalFontSize() {
				return 18;
			}
		};

		JFrame jframe = new JFrame();
		JediTermWidget tw = new JediTermWidget(settingsProvider);
		tw.getTerminal().setModeEnabled(TerminalMode.AutoNewLine, true);
		
		jframe.add(tw);
		
		jframe.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		jframe.setSize(1280, 720);
		jframe.setVisible(true);

		JediEmulator je = new JediEmulator(new ArrayTerminalDataStream(new char[0]), tw.getTerminal());
		OutputStream terminalOutputStream = new OutputStream() {
			@Override
			public void write(int ch) throws IOException {
				je.processChar((char) ch, tw.getTerminal());
			}
		};

		PrintStream terminalPrintStream = new PrintStream(terminalOutputStream);
		PrintStream stdout = System.out;
		TeeStream tee_stdout = new TeeStream(stdout, terminalPrintStream);
		System.setOut(tee_stdout);
		System.out.print(ESC + "%G"); //set to UTF-8 mode
		System.out.print(ESC + "[31;44m"); //colors don't change?
		System.out.println("HELLO");
		System.out.print(ESC + "[2J"); //nothing is erased?
		System.out.print(ESC + "[32;43m"); //colors don't change?
		System.out.println("TEST");
		System.out.flush();
		System.setOut(stdout);
		Thread.sleep(1000);
	}
}
```

**Comment by michael-newsrx** [comment]
While I'm sure this is not the "correct" way, I finally got the ANSI sequences being processed by using the following adjusted outputstream:

```Java
OutputStream terminalOutputStream = new OutputStream() {
			StringBuilder sb = new StringBuilder(4096);
			@Override
			public void write(int ch) throws IOException {
				sb.append((char)ch);
				if (ch == '\n') {
					flush();
				}
			}
			@Override
			public void flush() throws IOException {
				char[] charArray = sb.toString().toCharArray();
				sb.setLength(0);
				dataStream.pushBackBuffer(charArray, charArray.length);
				je.resetEof();
				while (je.hasNext()) {
					je.next();
				}
				super.flush();
			}
		};
```

**Comment by segrey** [comment]
@michael-newsrx Sorry for the delay and thanks for your example! Its updated version was merged in 8f605c41282df15ace46278beb1b6536f33bdb9d (see `com.jediterm.example.BasicTerminalExample`).

**Comment by michael-newsrx** [comment]
I still can't get this to work. How to I redirect System.out into the terminal here?
